common/lib/kconfig.ts

```diff
@@ -216,6 +217,76 @@ const ipfsGatewayUrl = (ipfs: string): string => `${ipfsGateway}${ipfsLinkToCid(
 const ipfsGatewayLink = (ipfs: string): string => urlToLink(ipfsGatewayUrl(ipfs), textShort(ipfs));
 /////////////////////////////////////////////////////////////////////////////////////////////////////
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// SWARM HELPERS
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////////////////////////
+// uri : https://ipfs.io/ipfs/bafkreieivwe2vhxx72iqbjibxabk5net4ah5lo3khekt6ojyn7cucek624
+// => ipfs uri : ipfs://bafkreieivwe2vhxx72iqbjibxabk5net4ah5lo3khekt6ojyn7cucek624
+/////////////////////////////////////////////////////////////////////////////////////////////////////
+const swarmGetLink = (uri: string | undefined): string => {
+  if (!uri) return "";
+
+  let swarmLink = "";
+  let cid = "";
+
+  if (uri.startsWith("swarm://")) {
+    swarmLink = uri;
+  } else if (uri.startsWith("https://api.gateway.ethswarm.org/bzz/")) {
+    // find cid in uri
+    cid = uri.replace("https://api.gateway.ethswarm.org/bzz/", "");
+
+    // console.log("ipfsGetLink ~ uri res cid", uri, res, cid);
+  }
+  if (cid) {
+    // reconstruct ipfs uri
+    swarmLink = swarmCidToLink(cid);
+  }
+
+  return swarmLink;
+};
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// http uri : https://api.gateway.ethswarm.org/bzz/1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+// ipfs uri : swarm://1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+// => gateway url : https://api.gateway.ethswarm.org/bzz/1fa18cf1aaee4727ecc266a86f1ef0f98b14771c7814d8cfb850a4b1c6d1359f
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+const swarmToUrlHttp = (url: string): string => (url.startsWith("swarm://") ? swarmGatewayUrl(url) : url);
```

Comment:
swarmToUrlHttp = (url: string)   => swarmLinkToUrlHttp = (link: string)

(rename ipfsToUrlHttp the same way ?)

Reply:
done